crates/ruff_dev/src/print_ast.rs

```diff
@@ -5,19 +5,26 @@ use std::fs;
 use std::path::PathBuf;
 
 use anyhow::Result;
-use rustpython_ast::Suite;
-use rustpython_parser::Parse;
+use rustpython_parser::{parse, Mode};
 
 #[derive(clap::Args)]
 pub(crate) struct Args {
     /// Python file for which to generate the AST.
     #[arg(required = true)]
     file: PathBuf,
+    /// Run in Jupyter mode i.e., allow line magics.
+    #[arg(long)]
+    jupyter: bool,
 }
 
 pub(crate) fn main(args: &Args) -> Result<()> {
     let contents = fs::read_to_string(&args.file)?;
-    let python_ast = Suite::parse(&contents, &args.file.to_string_lossy())?;
+    let mode = if args.jupyter {
```

Comment:
what about determining this from file extension, so you can run `print-ast` on `data_analysis.ipynb` without having to set the flag?

Reply:
Yes, that's helpful as well but this would allow us to use Jupyter specific syntax in a Python file and not worry about starting a notebook server. We could include both :)

Reply:
I'll probably add this in a later PR.